src/routes/transactions/mappers/common/human-description.mapper.ts

```diff
@@ -79,34 +80,95 @@ export class HumanDescriptionMapper {
+        if (fragment instanceof RichTokenValueFragment) {
+          return fragment.symbol
+            ? `${fragment.value} ${fragment.symbol}`
+            : fragment.value;
+        }
 
-            // Unlimited approval
-            if (fragment.value.amount === MAX_UINT256) {
-              return `unlimited ${token.symbol}`;
-            }
+        if (fragment instanceof RichAddressFragment) {
+          return this.shortenAddress(fragment.value);
+        }
+
+        return fragment.value;
+      })
+      .join(' ');
+  }
 
-            return `${formatUnits(fragment.value.amount, token.decimals)} ${
-              token.symbol
-            }`;
+  private async enrichFragments(
+    fragments: HumanDescriptionFragment[],
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<RichDecodedInfoFragment[]> {
+    return Promise.all(
+      fragments.map(async (fragment) => {
+        switch (fragment.type) {
+          case ValueType.TokenValue:
+            return this.enrichTokenValue(fragment, transaction, chainId);
           case ValueType.Address:
-            return this.shortenAddress(fragment.value);
-          default:
-            return fragment.value;
+            return new RichAddressFragment(fragment.value);
+          case ValueType.Text:
+            return new RichTextFragment(fragment.value);
+          case ValueType.Number:
+            return new RichTextFragment(fragment.value.toString());
         }
+      }),
+    );
+  }
+
+  private async enrichTokenValue(
+    fragment: TokenValueFragment,
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<RichTokenValueFragment> {
+    const token = await this.tokenRepository
+      .getToken({
+        chainId,
+        address: transaction.to,
       })
-      .join(' ');
+      .catch(() => null);
+
+    let amount: string;
+    if (fragment.value.amount === MAX_UINT256) {
+      amount = 'unlimited';
+    } else if (token && token.decimals) {
+      amount = formatUnits(fragment.value.amount, token.decimals);
+    } else {
+      amount = fragment.value.amount.toString();
+    }
+
+    return new RichTokenValueFragment(
+      amount,
+      token?.symbol ?? null,
+      token?.logoUri ?? null,
+    );
+  }
+
+  async enrichSafeAppInfo(
+    fragments: RichDecodedInfoFragment[],
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<RichDecodedInfoFragment[]> {
+    const safeAppInfo = isMultisigTransaction(transaction)
+      ? await this.safeAppInfoMapper.mapSafeAppInfo(chainId, transaction)
+      : null;
+
+    if (safeAppInfo) {
+      fragments.push(<RichTextFragment>{
+        type: RichFragmentType.Text,
+        value: `via ${safeAppInfo.name}`,
+      });
+    }
+
+    return fragments;
   }
 
-  shortenAddress(address: Hex, length = 4): string {
+  private shortenAddress(address: string, length = 4): string {
```

Comment:
Why was this one changed to `string`?

You can use `isHex` in the callers and keep the parameter as `Hex`.

Reply:
The `isAddress` call we do converts it to `Hex` thats why I simplified it here.
<img width="566" alt="Screenshot 2023-09-05 at 17 06 49" src="https://github.com/safe-global/safe-client-gateway/assets/5880855/3bc906e7-ed84-4535-99b6-818d3fd97ba3">

Reply:
Ok I was thinking of constraining the arguments instead of validating it inside but this is also ok from my side 👍